controller/src/beerocks/master/wfa_ca.cpp

```diff
@@ -669,3 +724,160 @@ bool wfa_ca::create_cmdu(ieee1905_1::CmduMessageTx &cmdu_tx, ieee1905_1::eMessag
 
     return true;
 }
+
+/**
+ * @brief Parse command "DEV_SEND_1905" TLVs parameters into an ordered list of TLV structure
+ * "tlv_hex_t", and does data validation.
+ * 
+ * @param[in,out] tlv_hex_list An ordered list of TLVs of type "tlv_hex_t".
+ * @param[in] params An unordered map containing parsed parameters of a CAPI command. 
+ * @param[out] err_string Contains an error description if the function fails.
+ * @return true if successful, false if not.
+ */
+bool wfa_ca::get_send_1905_1_tlv_hex_list(std::list<tlv_hex_t> &tlv_hex_list,
+                                          std::unordered_map<std::string, std::string> &params,
+                                          std::string &err_string)
+{
+    static const int max_tlv                          = 256; // Magic number
+    size_t tlv_idx                                    = 0;
+    bool skip                                         = false;
+    bool finish                                       = false;
+    static const std::vector<std::string> tlv_members = {"tlv_type", "tlv_length", "tlv_value"};
+
+    do {
+        tlv_hex_t tlv_hex;
+        for (size_t tlv_member_idx = 0; tlv_member_idx < tlv_members.size(); tlv_member_idx++) {
+            std::string lookup_str =
+                tlv_members[tlv_member_idx] + (tlv_idx > 0 ? std::to_string(tlv_idx) : "");
+            auto it = params.find(lookup_str);
+            if (it == params.end()) {
+                if (tlv_idx == 0 && tlv_member_idx == 0) {
+                    // Skipping TLV with no index (index 0), checking TLV index 1
+                    skip = true;
+                    break;
+                } else if (tlv_idx > 0 && tlv_member_idx == 0) {
+                    // No indexed TLVs found, finish
+                    finish = true;
+                    break;
+                } else {
+                    err_string = "missing param name '" + lookup_str + "'";
+                    return false;
+                }
+            }
+
+            // Type
+            if (tlv_member_idx == 0) {
+                tlv_hex.type = &it->second;
```

Comment:
We could actually do the conversion to `uint8_t` here, i.e. have uint8_t and uint16_t as the first two members of `tlv_hex_t`. Actually, we could even do the same for the value member and do the conversion immediately. Hm, but that would require creating a buffer in which to write first, so maybe not such a great idea...

Reply:
I prefer that this function will focus on parsing, and another one will do the copy to the buffer.